text/0000-core-builder.md

```diff
@@ -0,0 +1,46 @@
+
+Users will be able to use the new builder to build and run their applications using any of the language family buildpacks included. Instead of requireing a decision for an optimised builder, they can "just" start and later decide on an optimised `run` image.
+
+## Motivation
+
+The Paketo project will be easier to use for users - they can start without being asked to decide on optimisation.
+The stacks and builders maintainer team will have less work to do - they will only need to maintain a single stack and builder instead of three.
+
+## Detailed Explanation
+
+We would replace the current `base`, `tiny` and `static` stacks with a single stack that includes all packages required by the buildpacks. Similarly, we would replace the current `base`, `tiny` and `static` builders with a single builder.
+
+## Rationale and Alternatives
+
+We could leave everything as it is. We would stick to maintaining four stacks repositories and eight builder repositories and creating new ones for every LTS we adopt. But that would increase the burden on the maintainers.
+
+## Implementation
+
+I would propose to create the new stack and builder as part of moving to Ubuntu Noble Numbat. I.e. 
+- we should create a repository `paketo-buildpacks/stack-ubuntu-noble`
+- we should take the `build` image content from the `paketo-buildpacks/noble-base-stack` repository and apply it to `paketo-buildpacks/stack-ubuntu-noble`
+- we should take the `run` image content from the `paketo-buildpacks/noble-base-stack` repository and apply it to `paketo-buildpacks/stack-ubuntu-noble`
+- we should take the `run` image content from the `paketo-buildpacks/noble-tiny-stack` repository and apply it to `paketo-buildpacks/stack-ubuntu-noble` as additional run image
+- we should take the `run` image content from the `paketo-buildpacks/noble-static-stack` repository and apply it to `paketo-buildpacks/stack-ubuntu-noble` as additional run image
+- we should create a repository `paketo-buildpacks/builder-ubuntu-noble` and another`paketo-buildpacks/builder-ubuntu-noble-buildpackless`
+    - we should use the above `build` image as a base layer for the `builder` image
+    - we should use the above `run` image as the default `run` image for the `builder.toml`
+    - we should use the above `run-static` and `run-tiny` images as additional `run` images for the `builder.toml`
+
+I would propose to delete the `paketo-buildpacks/noble-base-stack`, `paketo-buildpacks/noble-tiny-stack` and `paketo-buildpacks/noble-static-stack` repositories.
+I would propose to delete the `paketo-buildpacks/builder-noble-buildpackless-static`, `paketo-buildpacks/builder-noble-buildpackless-tiny` and `paketo-buildpacks/builder-noble-buildpackless-base` repositories.
+
+## Prior Art
+
+None.
+
+## Unresolved Questions and Bikeshedding
+
+What about the `full` stack and builders?
```

Comment:
I feel like we should include full in this proposal. If we don't, that would effectively mean dropping support for buildpacks that only work in the full builder.

Have you looked at the difference between the base build image and the full build image? If it's not too much of a difference, we could conceivably do the same thing and just have another run image for full.

Reply:
I haven't looked into this. I'd hesitate to increase the size and attack surface of the builder based on the needs of a few buildpacks. The build image is by far not as critical as the run images, but it still feels like a lot like catering for edge cases and put the weight on standard cases.

We should probably just keep providing a full builder in addition to the one proposed here. WDYT?

Reply:
The [list of packages in full](https://github.com/paketo-buildpacks/jammy-full-stack/blob/6f7fccb60993efc39b78008300b03b9ac15c5ecc/stack/stack.toml#L21-L202) is quite a lot larger than [the list of packages in base](https://github.com/paketo-buildpacks/jammy-base-stack/blob/ce49ff79924e71eb0d55c75f3c96cc3af730226e/stack/stack.toml#L21-L38). 

For me, adding full packages to this new build image is not an option. I guess I will put a note into the RFC to show that if we need a full build image, we need separate build, run and builder images.

Reply:
```suggestion
Q: What about the `full` stack and builders?
A: This proposal does not include the `full` builder and recommends that it be deprecated. If approved, this RFC will make the Jammy Full stack the last supported iteration of the full stack. There is presently only one buildpack, PHP, which requires the full stack and is thus impacted. This buildpack will be able to continue using the Jammy Full stack through [April 2027](https://ubuntu.com/about/release-cycle) with the goal of converting it to run on the new Noble all-in-one stack before that point.
```

Ugh, yes. That's a lot of stuff we'd have to add. I agree that it's too much to stuff all in one build image. It's also a lot of work to support one additional buildpack. How about this?